ewah/hooks/shopify.py

```diff
@@ -261,11 +295,22 @@ def get_data(
                     )
             yield data
 
-            if response.headers.get("Link") and response.headers["Link"].endswith(
-                'el="next"'
+            if (
+                response.headers.get("Link")
+                and response.headers["Link"].endswith('el="next"')
             ):
                 self.log.info("Requesting next page of data...")
                 url = response.headers["Link"][1:-13]
+            elif ids_list and shopify_object == 'inventory_levels':
+                # after pagination complete we restart the requests while
+                # we still have ids in id_list
+                finished_pagination = True
+                url = self._BASE_URL.format(
+                    shop=shop_id,
+                    version=version,
+                    object="inventory_levels",
+                )
```

Comment:
Comment (non-blocking): This should not be necessary. Can you check?

Reply:
Lets discuss

Reply:
As discussed, my comment is invalid and your code is correct as is :)